cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -1291,6 +1358,14 @@ data TxReturnCollateral ctx era where
 deriving instance Eq   (TxReturnCollateral ctx era)
 deriving instance Show (TxReturnCollateral ctx era)
 
+instance IsMaybeLike (TxReturnCollateral ctx era) where
```

Comment:
I think rather than introducing the `IsMaybeLike` typeclass we should just bias the `Semigroup` instance as follows:
```
instance Semigroup (TxReturnCollateral ctx era) where
  x <> TxReturnCollateralNone = x
  TxReturnCollateralNone <> x = x
  _ <> TxReturnCollateral wit txOutB = TxReturnCollateral wit txOutB
```

Reply:
Done